### ISSUE TITLE
Framerate cap is higher in game than set value, and frametimes under cap are very inconsistent.

### ISSUE THREAD

**Issue by RahuX** [34736384/genshin-fps-unlock#240]
For reference, the current observations were made using the global version of Special K to display both the framerate and frametimes. More because I just use Special K to resolve some weird flip model behavior Genshin exhibits in certain menus without it.

As an example of the first point, setting a framerate cap to 120, the reported framerate in game is actually a bit higher than the set value, resulting more in a cap of 121 or 122 FPS. At a value of 144, it tends to hover closer to 145. Not a massive deal as you can just set the FPS cap a couple of frames lower than the desired target to avoid things like tearing/input lagged vsync kicking in, but still a minor annoyance.

For the second point, I'll post some examples using Special K's UI and built in framerate limiter for comparison points. The game has settings set low enough to hit, and is in an area that is consistently hitting, the target framerate (120 FPS in this case).

FPS Unlocker set to 120, Special K limiter disabled:
![Untitled](https://github.com/34736384/genshin-fps-unlock/assets/5471553/0b77325c-eb94-4e7f-b9e9-52363baaa3fb)

FPS Unlocker set to 120 or 144 (doesn't matter, same result), Special K limiter enabled & set to 120:
![Untitled2](https://github.com/34736384/genshin-fps-unlock/assets/5471553/6ecc256e-6cfb-4834-ad5f-1c1175a96993)

Usually using 2 forms of framerate caps at the same time is generally not advisable, but since the framerate cap with just the FPS Unlocker produces inconsistent frametimes (which just further exacerbates issues like #215 that cropped up with Fontaine), I feel like I have no choice but to do so.

Maybe this is more down to the game's (so Unity's) built in framerate limiter producing these results, thus nothing can be done, but I figured I'd would just bring this up just in case it is fixable at all.

**Comment by Spedemix** [comment]
Yeah, so usually unlocking the frame rate will result in pretty inconsistent frame times in this game. So far i've just come to expect it.

Have you tried other forms of frame limiting like using the default frame rate limiter in your GPUs control panel? Or is specifically Special K's fps limiter that gives you smoother frame times? Also, some of the stuttering in the game goes beyond the frame time (different aspects of the game updating at different intervals) so do you also feel that the game runs smoother when you use Special K's fps limiter? Does the SK limiter also help with the stuttering in some of the Fontaine regions?

**Comment by RahuX** [comment]
> Yeah, so usually unlocking the frame rate will result in pretty inconsistent frame times in this game. So far i've just come to expect it.
> 
> Have you tried other forms of frame limiting like using the default frame rate limiter in your GPUs control panel? Or is specifically Special K's fps limiter that gives you smoother frame times? Also, some of the stuttering in the game goes beyond the frame time (different aspects of the game updating at different intervals) so do you also feel that the game runs smoother when you use Special K's fps limiter? Does the SK limiter also help with the stuttering in some of the Fontaine regions?

Just tested with the NVIDIA Control Panel cap set to 120, Special K's off. Also gives me smooth results:

![image](https://github.com/34736384/genshin-fps-unlock/assets/5471553/3100fae4-b0ae-4e62-91fd-ba706d3036ab)

As for feel, it certainly feels a bit smoother than with just the unlocker by itself. Might not be extremely noticeable, but I tend to be a bit more sensitive to frametimes.

In regards to stuttering, true the game does have natural stuttering due to loading in new areas, so it won't alleviate that, but general traversal/traversal stutter does feel and appear smoother.

It won't help with the recent frametime oddities that have cropped up regarding going underwater though. Only thing that seems to help there is reducing render res, or disabling GI.

**Comment by Spedemix** [comment]
Huh, well then it really seems like the frame rate limiter in the unlocker itself is of worse quality than whatever Nvidia or SpecialK are using. I guess I'll have to test it myself as well.

**Comment by IphonyIvy** [comment]
I do not know if this is not related to this error also I dont know if this might help.

But I have noticed after genshin update 4.2 they game constantly crashes with the following error (Idk if anyone else has gotten it)
![Crash](https://github.com/34736384/genshin-fps-unlock/assets/151812178/03b52e53-4e12-4132-b8b6-bd5eb0039adf)

But, this only happens if you are running an NVDIA driver higher than 537.58.  If you run driver 537.58 with the fps unlocker everything runs super smooth and fine.  If I go to a higher version, I do get the blu-screen with the error.

If I use any of the newer versions without FPS unlocker, it does not crash.

This could be a "me" issue and not help what so ever and Im sorry for wasting your time. I just thought this could be associated with the current given issue as its related to fps in general.
